msal-dotnet-articles/acquiring-tokens/desktop-mobile/username-password-authentication.md

```diff
@@ -1,271 +1,260 @@
-   // Explanation: this can happen if your application was not registered as a public client application in Azure AD
-   // Mitigation: in the Azure portal, edit the manifest for your application and set the `allowPublicClient` to `true`
-   // ------------------------------------------------------------------------
-  }
-  catch (MsalServiceException)
-  {
-   throw;
-  }
-
-  catch (MsalClientException ex) when (ex.ErrorCode == "unknown_user_type")
-  {
-   // Message = "Unsupported User Type 'Unknown'. Please see https://aka.ms/msal-net-up"
-   // The user is not recognized as a managed user, or a federated user. Azure AD was not
-   // able to identify the IdP that needs to process the user
-   throw new ArgumentException("U/P: Wrong username", ex);
-  }
-  catch (MsalClientException ex) when (ex.ErrorCode == "user_realm_discovery_failed")
-  {
-   // The user is not recognized as a managed user, or a federated user. Azure AD was not
-   // able to identify the IdP that needs to process the user. That's for instance the case
-   // if you use a phone number
-   throw new ArgumentException("U/P: Wrong username", ex);
-  }
-  catch (MsalClientException ex) when (ex.ErrorCode == "unknown_user")
-  {
-   // the username was probably empty
-   // ex.Message = "Could not identify the user logged into the OS. See http://aka.ms/msal-net-iwa for details."
-   throw new ArgumentException("U/P: Wrong username", ex);
-  }
-  catch (MsalClientException ex) when (ex.ErrorCode == "parsing_wstrust_response_failed")
-  {
-   // ------------------------------------------------------------------------
-   // In the case of a Federated user (that is owned by a federated IdP, as opposed to a managed user owned in an Azure AD tenant)
-   // ID3242: The security token could not be authenticated or authorized.
-   // The user does not exist or has entered the wrong password
-   // ------------------------------------------------------------------------
-  }
- }
-
- Console.WriteLine(result.Account.Username);
+    string authority = "https://login.microsoftonline.com/contoso.com";
+    string[] scopes = new string[]
+    {
+        "user.read"
+    };
+
+    IPublicClientApplication app;
+    app = PublicClientApplicationBuilder.Create(clientId).WithAuthority(authority).Build();
+
+    var accounts = await app.GetAccountsAsync();
+    AuthenticationResult result = null;
+
+    if (accounts.Any())
+    {
+        result = await app.AcquireTokenSilent(scopes, accounts.FirstOrDefault()).ExecuteAync();
+    }
+    else
+    {
+        try
+        {
+            var securePassword = new SecureString();
+            foreach(char c in "dummy") // you should fetch the password keystroke
+            securePassword.AppendChar(c); // by keystroke
+            result = await app.AcquireTokenByUsernamePassword(scopes, "joe@contoso.com", securePassword).ExecuteAsync();
+        }
+        catch (MsalUiRequiredException ex) when(ex.Message.Contains("AADSTS65001"))
```

Comment:
Can we just show this and remove other duplicate code that's in the above snippet? Or combine?